watch.go

```diff
@@ -21,7 +23,23 @@ type Event struct {
 	// Path to the data that changed
 	Path string
 	// Data that changed
-	Data interface{}
+	Data    interface{}
+	payload string
+}
+
+func (e Event) Value(v interface{}) (path string, err error) {
```

Comment:
In order for us to see those gains though, we would have to remove the old style. For the time being, I think we should return an error from this function and just have the consumer access the path through the `Event` struct.